### PR TITLE
Create workflows filtering Autcompletion component

### PR DESCRIPTION
**Pull request by feedmeapples**
This component can suggest and autocomplete the Query that user is typing, the following is currently supported:
- looks at the current cursor and relevant chunk to autocomplete
- understands the type of the chunk to autocomplete: `search attribute`, `logical operator`, `relation operator`, `value`
- suggest the relevant options for autocompletion, such as search attributes, operators, a value for some types of search attributes, such as a current Date value for search attributes of type datetime, etc.  

Next to add to this component:
Improved UX, add full keyboard navigation, improved styling

#### Filtering and suggesting search attributes:
image

#### Relational operators:
image

#### Values:
image

#### Logical operators:
image

#### Shows suggestions according to the current cursor location, in this case it's the attribute "RunId"
image

#### When filtering search attributes by empty value
image

### CODE REVIEW

**Review by sw-yx — APPROVED** [review]
LGTM. what is your test plan for this? also make sure to check the tab order e.g. if i tab or hit enter does the autocomplete interfere with my intuition. 

i also think keyboard navigation (down/up) is impt here. see if theres a premade library we can use. in React we have downshift, theres probably a similar one for Vue.

**Comment by feedmeapples** [comment]
> LGTM. what is your test plan for this? also make sure to check the tab order e.g. if i tab or hit enter does the autocomplete interfere with my intuition.
> 
> i also think keyboard navigation (down/up) is impt here. see if theres a premade library we can use. in React we have downshift, theres probably a similar one for Vue.

I will create PRs for these improvements subsequently:
Improved UX, add full keyboard navigation, improved styling

File: client/components/workflow-filter-autocomplete/constants.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+};
+
+export const OPERATORS_LOGICAL_LIST = Object.entries(OPERATORS_LOGICAL).map(
+  ([key, value]) => value
+);
+
+export const SUGGESTION_STATES = {
+  selectSearchAttribute: 0,
+  selectOperatorRelational: 1,
+  selectOperator: 1,
```
Comment (feedmeapples) [inline client/components/workflow-filter-autocomplete/constants.js:28]: leftover

File: client/components/workflow-filter-autocomplete/index.vue [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+  .suggestion {
+    list-style: none;
+    text-align: left;
+    padding: 4px 2px;
+    cursor: pointer;
+  }
+
+  .suggestion:hover {
+    background-color: temporal-purple;
+    // color: white;
```
Comment (feedmeapples) [inline client/components/workflow-filter-autocomplete/index.vue:213]:
```suggestion
```

File: client/components/workflow-filter-autocomplete/index.vue [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+      let [start, end] = [cursorPos, cursorPos];
+      while (start > 0 && this.query[start - 1] != ' ') {
+        start--;
+      }
+      while (end < this.query.length && this.query[end] != ' ') {
+        end++;
+      }
+
+      let word = this.query.slice(start, end);
```
Comment (feedmeapples) [inline client/components/workflow-filter-autocomplete/index.vue:157]:
```suggestion
      const word = this.query.slice(start, end);
```

File: client/components/workflow-filter-autocomplete/constants.js (hunk 1 above)
Comment (feedmeapples) [inline client/components/workflow-filter-autocomplete/constants.js:28]:
```suggestion
```

**Review by rylandg — APPROVED** [review]
Small nits:

Overall this is the direction Temporal web needs to go.

**Comment by sw-yx** [comment]
fwiw as of the snapshot from https://hub.docker.com/r/markmark206/tweb its flickering every time i type a query


image

**Comment by feedmeapples** [comment]
> fwiw as of the snapshot from https://hub.docker.com/r/markmark206/tweb its flickering every time i type a query
> 
> image

i don't think markmark206/tweb has the image for this pull request. Will ping you in dm to discuss/demo + decide on proper feature specification

**Comment by sw-yx** [comment]
This PR is on hold pending internal agreement on feature requirements. [internal notes](https://www.notion.so/temporalio/PRD-for-workflows-filter-autocompletion-1cbc065bcca74aa5bc03fd34ad1b15b2)

**Pull request closed without merging**